### DISCUSSION TITLE
Row._asdict

### DISCUSSION THREAD

**Discussion by mehdigmira** [sqlalchemy/sqlalchemy#10091] [Ideas]
Sqalchemy row objects have an _asdict method that allows transforming the row to a dict:  https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Row._asdict

I understand this is to emulate namedtuple interface, but would it be possible to expose it as a public method without the "_" prefix ? I think exposing private methods is a bad pattern

**Comment by zzzeek** [comment]
`row` is an named tuple, so the full namespace of non-underscored items is left open.

We would not have added `_asdict()` with the private underscore if Python standard lib didn't also do it.   If Python standard lib added a non-underscored `_asdict` to namedtuple, we would add it also.

Maybe post a bug with Python then?  I'm sure this must have been discussed

**Reply by CaselIT** [reply]
Also you may look into https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.mappings if you want to a result that returns dict like objects.

**Reply by zzzeek** [reply]
it's also pointed out that we probably should rename the `.tuple()` and `.t` accessors which aren't following this pattern right now on `Row` and I agree we should do that.

**Reply by CaselIT** [reply]
created https://github.com/sqlalchemy/sqlalchemy/issues/10093 to track that

**Reply by mehdigmira** [reply]
didn't know about mappings, thx

**Reply by firezdog** [reply]
Yeah it would be nice if this were changed or a proxy method were created.  The "_" strongly implies that the method is private and that there is some other favored way of converting a row to a dictionary.

**Reply by CaselIT** [reply]
Row uses the namedtuple interface. the name is taken from the stdlib
